datasets/amazon_us_reviews/amazon_us_reviews.py

```diff
@@ -0,0 +1,199 @@
+    _DATA_OPTIONS.append(entry + "_v1_01")
+
+for entry in _DATA_OPTIONS_V1_02:
+    _DATA_OPTIONS.append(entry + "_v1_02")
+
+_DL_URLS = {
+    name: "https://s3.amazonaws.com/amazon-reviews-pds/tsv/amazon_reviews_us_" + name + ".tsv.gz"
+    for name in _DATA_OPTIONS
+}
+
+
+class AmazonUSReviewsConfig(datasets.BuilderConfig):
+    """BuilderConfig for AmazonUSReviews."""
+
+    def __init__(self, *, data=None, **kwargs):
+        """Constructs a AmazonUSReviewsConfig.
+        Args:
+            data: `str`, one of `_DATA_OPTIONS`.
+            **kwargs: keyword arguments forwarded to super.
+        """
+        if data not in _DATA_OPTIONS:
+            raise ValueError("data must be one of %s" % _DATA_OPTIONS)
+
+        super(AmazonUSReviewsConfig, self).__init__(**kwargs)
+        self.data = data
+
+
+class AmazonUSReviews(datasets.GeneratorBasedBuilder):
+    """AmazonUSReviews dataset."""
+
+    BUILDER_CONFIGS = [
+        AmazonUSReviewsConfig(  # pylint: disable=g-complex-comprehension
+            name=config_name,
+            description=(
+                f"A dataset consisting of reviews of Amazon {config_name} products in US marketplace. Each product "
+                "has its own version as specified with it."
+            ),
+            version=datasets.Version("0.1.0", ""),
+            data=config_name,
+        )
+        for config_name in _DATA_OPTIONS
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "marketplace": datasets.Value("string"),
+                    "customer_id": datasets.Value("string"),
+                    "review_id": datasets.Value("string"),
+                    "product_id": datasets.Value("string"),
+                    "product_parent": datasets.Value("string"),
+                    "product_title": datasets.Value("string"),
+                    "product_category": datasets.Value("string"),
+                    "star_rating": datasets.Value("int32"),
+                    "helpful_votes": datasets.Value("int32"),
+                    "total_votes": datasets.Value("int32"),
+                    "vine": datasets.features.ClassLabel(names=["Y", "N"]),
+                    "verified_purchase": datasets.features.ClassLabel(names=["Y", "N"]),
+                    "review_headline": datasets.Value("string"),
+                    "review_body": datasets.Value("string"),
+                    "review_date": datasets.Value("string"),
+                }
+            ),
+            supervised_keys=None,
+            homepage="https://s3.amazonaws.com/amazon-reviews-pds/readme.html",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        url = _DL_URLS[self.config.name]
+        path = dl_manager.download_and_extract(url)
+
+        # There is no predefined train/val/test split for this dataset.
+        return [
```

Comment:
I'm fine with this! I did the splitting myself for other datasets, just (90%, 5%, 5%), but not at all sure what the better approach is here. Think this way here is actually better because if there is no official split we should not do it and the user can easily make use of `select` or `filter` to do the split her/himself.

Reply:
Yeah my feeling is if there's no official split that it's best to let the user create their own, but I guess that's a general question that we don't have a consensus around.